forge-gui/res/cardsfolder/upcoming/sinuous_benthisaur.txt

```diff
@@ -0,0 +1,11 @@
+Name:Sinuous Benthisaur
+ManaCost:5 U
+Types:Creature Dinosaur
+PT:4/4
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDig | TriggerDescription$ When CARDNAME enters the battlefield, look at the top X cards of your library, where X is the number of Caves you control plus the number of Cave cards in your graveyard. Put two of those cards into your hand and the rest on the bottom of your library in a random order.
+SVar:TrigDig:DB$ Dig | DigNum$ X | ChangeNum$ 2 | RestRandomOrder$ True
+SVar:X:Count$Valid Cave.YouCtrl/Plus.Y
```

Comment:
I thought about that but doesn't YouCtrl vs YouOwn matter?

Or just use YouCtrl?

Reply:
rules are on your side:
> 108.4a. If anything asks for the controller of a card that doesn't have one (because it's not a permanent or spell), use its owner instead.
but just a matter of taste